### PR TITLE
[risk=no][RW-6769] Source-info component & function

### PR DESCRIPTION
**Pull request by mhigbyflowers**

### CODE REVIEW

File: public-ui/src/app/data-browser/components/source-tree/source-tree-react.component.tsx [hunk 1]
```diff
@@ -59,32 +59,21 @@ export const SourceTreeComponent = (
...
-            this.setState({
-                isConceptSelected: !this.state.isConceptSelected
-            });
-            this.getTreeHighlight();
-        }
-        getTreeHighlight() {
-            this.setState({
-                highlightId: parseInt(localStorage.getItem('treeHighlight'), 10)
-            });
+            localStorage.setItem('selectedTreeNode', JSON.stringify(this.props.node));
```
Comment (mhigbyflowers) [inline public-ui/src/app/data-browser/components/source-tree/source-tree-react.component.tsx]: This is where the data should bubble up to the parent. Local Storage doesn't seem like the best solution.

File: public-ui/src/app/data-browser/views/concept-chart/concept-chart-react.component.tsx [hunk 2]
```diff
@@ -291,108 +315,152 @@ export class ConceptChartReactComponent extends React.Component<Props, State> {
...
+                treeData = result.parent;
+                // treeLoading = false;
+                this.setState({
+                    node: treeData
+                });
+            });
+    }
+
+    childConceptClicked() {
+        const selectedNode = JSON.parse(localStorage.getItem('selectedTreeNode'));
```
Comment (mhigbyflowers) [inline public-ui/src/app/data-browser/views/concept-chart/concept-chart-react.component.tsx]: This is where the data is picked up in the parent

File: public-ui/src/app/data-browser/views/concept-chart/concept-chart-react.component.tsx [hunk 3]
```diff
@@ -291,108 +317,152 @@ export class ConceptChartReactComponent extends React.Component<Props, State> {
...
+    loadSourceTree(concept: any) {
+        let treeData;
+        dataBrowserApi().getCriteriaRolledCounts(concept.conceptId, 'condition')
+            .then(result => {
+                treeData = result.parent;
+                // treeLoading = false;
+                this.setState({
+                    node: treeData
+                });
+            });
```
Comment (SrushtiGangireddy) [inline public-ui/src/app/data-browser/views/concept-chart/concept-chart-react.component.tsx:329]: dataBrowserApi().getCriteriaRolledCounts(concept.conceptId, 'condition')
            .then(result => {
                // treeLoading = false;
                this.setState({
                    node: result.parent
                });
            });

File: public-ui/src/app/data-browser/views/concept-chart/concept-chart-react.component.tsx [hunk 4]
```diff
@@ -291,108 +317,152 @@ export class ConceptChartReactComponent extends React.Component<Props, State> {
...
+        const id = parseInt(selectedNode.conceptId, 10);
+        dataBrowserApi().getSourceConcepts(id)
+            .then(results => {
+                this.setState({
+                    selectedTreeNode: selectedNode,
+                    selectedTreeConcept: id
+                });
+            }).catch(e => {
+                console.log(e, 'error');
+            });
```
Comment (SrushtiGangireddy) [inline public-ui/src/app/data-browser/views/concept-chart/concept-chart-react.component.tsx:342]: dataBrowserApi().getSourceConcepts(id)
            .then(results => {
                this.setState({
                    selectedTreeNode: selectedNode,
                    selectedTreeConcept: parseInt(selectedNode.conceptId, 10)
                });
            }).catch(e => {
                console.log(e, 'error');
            });

**Review by SrushtiGangireddy — APPROVED** [review]
Looks good to me.

**Pull request merged**